src/Compilers/CSharp/Portable/Symbols/Source/SourceAssemblySymbol.cs

```diff
@@ -1360,49 +1399,52 @@ private void LoadAndValidateNetModuleAttributes(ref CustomAttributesBag<CSharpAt
                 {
                     wellKnownData = ValidateAttributeUsageAndDecodeWellKnownAttributes(attributesFromNetModules, netModuleNames, diagnostics);
```

Comment:
>wellKnownData = ValidateAttributeUsageAndDecodeWellKnownAttributes(attributesFromNetModules, netModuleNames, diagnostics); [](http://example.com/codeflow?start=20&length=122)

Are we testing this codepath? It looks like this function is going to call ```GetUniqueSourceAssemblyAttributes``` unconditionally.

Reply:
It feels to me that, instead of modifying this method and trying to find and suppress all "dangerous" things it might do, it might be simpler/cleaner to create a new helper that does exactly what we want. We can easily test that it does the right thing for the three attributes we care about.

Reply:
No, I'm not testing the scenario with netmodule attributes. You're correct there's still a circularity on that path, but that's beyond the purpose of this mitigation/PR. We're not closing all possible circularities we're trying to avoid the common one.

Reply:
"noCache" tells nothing about us wanting to prevent just common circularities. Users will have certain expectations, which are likely not met with the current approach

Reply:
How about `reduceCircularity`?

Reply:
> How about reduceCircularity?

I bel;ieve "it might be simpler/cleaner to create a new helper that does exactly what we want. We can easily test that it does the right thing for the three attributes we care about" I really believe it will be simple to extract the code that goes through netModules' attributes building result that we want without reporting any diagnostics etc.. No additional flags necessary.

Reply:
I'll give this a shot because I trust your judgement, but I will timecap it in case it doesn't turn out simple.